### PR TITLE
[SDK] Fix build for all targets

### PR DESCRIPTION
**Pull request by br41nslug**
The initial migration simply relied on `tsc` for a basic build. This PR re-instates the original `rollup` build configuration for `esm`, `cjs` and `iife` targets.

Follow up to #18647

> Note: An attempt at using `tsup` was made but aborted due to lack of flexibility.

### CODE REVIEW

File: packages/sdk/tsup.config.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		switch (format) {
+			case 'esm':
+				name = `.esm.min.js`;
+				break;
+			case 'cjs':
+				name = `.cjs.js`;
+				break;
+			case 'iife':
+				name = `.bundler.js`;
+				break;
```
Comment (rijkvanzanten) [inline packages/sdk/tsup.config.js:24]: Should we update these to:

```js
index.mjs // esm
index.cjs // cjs
index.js  // agnostic iife ?
```

?

Comment (br41nslug) [reply packages/sdk/tsup.config.js:24]: I am not sure if all runtimes accept the `mjs` and `cjs` extensions 🤔 but if they do, definitely

Comment (rijkvanzanten) [reply packages/sdk/tsup.config.js:24]: Ah yeah fair. My Node 20 defaultism privilege is showing :D

**Review by nickrum — APPROVED** [review]
We should probably modernize the rollup config at some point by e.g. using `rollup-plugin-esbuild` instead of `rollup-plugin-typescript2` and `@rollup/plugin-terser` instead of `rollup-plugin-terser`. Also, right now the node ESM import points to `index.mjs` which in turn imports to cjs entrypoint. We might want to add a dedicated node ESM target to the rollup config.

But I think this is good enough for the time being 👍

File: packages/sdk/package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
 		"node",
 		"sdk"
 	],
+	"type": "module",
```
[the remark this answers is not in the document]
Comment (nickrum) [reply packages/sdk/package.json:15]: Yes, you are right. We should either remove that or change the file extension of the cjs entrypoint to `.cjs` 👍

File: packages/sdk/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import json from '@rollup/plugin-json';
+import resolve from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import typescript from 'rollup-plugin-typescript2';
```
Comment (paescuj) [inline packages/sdk/rollup.config.js:4]: Let's use the official TypeScript plugin: https://www.npmjs.com/package/@rollup/plugin-typescript

Comment (nickrum) [reply packages/sdk/rollup.config.js:4]: The official Typescript plugin has some weird quirks (unfortunately I can't remember the exact details), which is why I went with `rollup-plugin-typescript2` back in the days. If we are going to touch this, I think we should go directly with something like `rollup-plugin-esbuild`.

File: packages/sdk/rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import sourceMaps from 'rollup-plugin-sourcemaps';
```
[the remark this answers is not in the document]
Comment (nickrum) [reply packages/sdk/rollup.config.js:5]: Yeah, I'm fairly certain this was for including `axios` sourcemaps!

[the remark this answers is not in the document]
Comment (paescuj) [reply packages/sdk/rollup.config.js:5]: Okay, let's drop it in that case, as we're dropping `axios` as well 👍

[the remark this answers is not in the document]
Comment (br41nslug) [reply packages/sdk/rollup.config.js:5]: Thats very good to know, i'll remove that in https://github.com/directus/directus/pull/18711 where we are actually dropping `axios`

**Pull request closed without merging**